homeassistant/helpers/config_validation.py

```diff
@@ -1325,11 +1326,30 @@ def script_action(value: Any) -> dict:
     }
 )
 
+
+def _backward_compat_service_schema(value: Any | None) -> Any:
+    """Backward compatibility for service schemas."""
+
+    if not isinstance(value, dict):
+        return value
+
+    # `service` has been renamed to `action`
+    if CONF_SERVICE in value:
+        if CONF_ACTION in value:
+            raise vol.Invalid(
```

Comment:
Please add a test for this, nothing fancy, just add a hands on test of `SERVICE_SCHEMA` to `tests/helpers/test_config_validation.py` which triggers the error

Reply:
🤦 Not sure why I missed that one.

Reply:
Added coverage.